packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -52,54 +51,96 @@ const toJsonBKey = <T extends keyof NestedFilterQueryFields>(nestedField: T): Js
+  '=@': 'ilike' as const, // Contains sub string
+};
+type Operator = keyof typeof operatorToSqlComparator;
+
+const filterOperators = Object.keys(operatorToSqlComparator) as Operator[];
+
+const formatComparisonValue = (value: string | string[], operator: Operator) => {
+  if (operator === '=@' && typeof value === 'string') {
+    return `%${value}%`;
   }
 
-  const [field, value] = clauseParts;
-  if (!isFilterableField(field)) {
-    throw new Error(`Unknown filter key: ${field}, must be one of: ${filterableFields}`);
+  return value;
+};
+
+const convertValueToAdvancedCriteria = (operator: Operator, value: string | string[]) => {
+  // For equal operator, we do not need to specify comparison object.
+  if (operator === '==') {
+    return value;
   }
 
-  const formattedField = isNestedField(field) ? toJsonBKey(field) : field;
-  return [formattedField, value] as [typeof formattedField, typeof value];
+  const comparator = operatorToSqlComparator[operator];
+
+  return {
+    comparator,
+    comparisonValue: formatComparisonValue(value, operator),
+  };
 };
 
-const buildCountryCodeFilter = (allowedCountries: string[], value?: string | string[]) => {
-  if (!value) {
-    return allowedCountries;
+const toFilterClause = (queryClause: string) => {
+  const operator = filterOperators.find(o => queryClause.includes(o));
+
+  if (!operator) {
+    throw new Error(
+      `Invalid query clause: '${queryClause}'. Cannot find valid operator. Available operators are: ${filterOperators.toString()}`,
+    );
   }
 
-  const filteredAllowedCountries = Array.isArray(value)
-    ? allowedCountries.filter(country => value.includes(country))
-    : allowedCountries.filter(country => value === country);
+  const clauseParts = queryClause.split(operator);
+  if (clauseParts.length !== 2) {
+    throw new Error(
+      `Invalid query clause: '${queryClause}'. Filter clause must be of format: <field><operator><value>`,
+    );
+  }
 
-  if (filteredAllowedCountries.length < 1) {
-    throw new Error('No access to any countries due to user permissions and country_code filter');
+  const [field, value] = clauseParts;
+  if (!isFilterableField(field)) {
+    throw new Error(`Unknown filter key: ${field}, must be one of: ${filterableFields}`);
   }
 
-  return filteredAllowedCountries;
+  const formattedField = isNestedField(field) ? toJsonBKey(field) : field;
+
+  return [formattedField, operator, value] as [
+    typeof formattedField,
+    typeof operator,
+    typeof value,
+  ];
 };
 
-export const extractFilterFromQuery = (allowedCountries: string[], queryFilter?: string) => {
+export const extractFilterFromQuery = (
+  allowedCountries: string[],
+  queryFilter?: string,
+): EntityFilter => {
   if (!queryFilter) {
-    return { country_code: allowedCountries };
+    // default filter to only get entities in allowed countries.
+    return {
+      [QueryConjunctions.AND]: {
+        country_code: allowedCountries,
+      },
```

Comment:
What do you think about return this as default if there's no filter instead of the previous:
```
{ country_code: allowedCountries }
```

When it was returning the previous format, I found it hard to debug the country_code filter because I kinda got confused a lot regarding whether country_code is something from the query filter or is it the default allowed countries. So I just change it to return this format (consistent with how we have the default filter when merging at the end of this function). So when I see this, then I know it's the default allowed countries filter.